include/crpropa/Grid.h

```diff
@@ -340,217 +296,183 @@ T interpolate(const Vector3d &position) {
 		return get(ix, iy, iz);
 	}
 
-	
+
+
 private:	
 
-__m128 simd_periodicGet(size_t ix, size_t iy, size_t iz) const {
+  __m128 simdPeriodicGet(size_t ix, size_t iy, size_t iz) const {
 		ix = periodicBoundary(ix, Nx);
 		iy = periodicBoundary(iy, Ny);
 		iz = periodicBoundary(iz, Nz);
-		return convert_Vector3d_to_simd(grid[ix * Ny * Nz + iy * Nz + iz]);
+		return convertVector3fToSimd(grid[ix * Ny * Nz + iy * Nz + iz]);
 	}
 
-__m128 convert_Vector3d_to_simd(const Vector3d v) const
-	{
-		__m128 sim_d_var = _mm_set_ps(0,v.z,v.y,v.x); 
-		return sim_d_var;
+  __m128 convertVector3fToSimd(const Vector3f v) const {
+		__m128 simdVar = _mm_set_ps(0,v.z,v.y,v.x);
+		return simdVar;
 	}
 	
-Vector3d convert_simd_to_Vector3d(__m128 res) const
-	{
+  Vector3d convertSimdToVector3f(__m128 res) const {
 		float vec[4];	
 		_mm_store_ps(&vec[0], res);
-		Vector3d result = Vector3d(vec[0], vec[1], vec[2]);
+		Vector3f result = Vector3d(vec[0], vec[1], vec[2]);
```

Comment:
The constructor on the right side should probably be Vector3f as well.

Reply:
Yes, that's right!

Reply:
Thanks for fixing! Did you try to build it or should I try to? I have a feeling that this may propagate outwards, at least into the return type of tricubicInterpolate(Vector3f, ...).